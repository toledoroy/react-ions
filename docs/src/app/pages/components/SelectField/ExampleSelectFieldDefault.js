import React from 'react'
import SelectField from 'react-ions/lib/components/SelectField/SelectField'

const options = [
  {value: '0', display: 'test 1', someOtherProp: true},
  {value: '1', display: 'test 2', someOtherProp: false},
  {value: '2', display: 'test 3', someOtherProp: false},
  {value: '3', display: 'test 4', someOtherProp: false},
  {value: '4', display: 'test 5', someOtherProp: false},
  {value: '5', display: 'test 6', someOtherProp: false},
  {value: '6', display: 'test 7', someOtherProp: false},
  {value: '7', display: 'test 8', someOtherProp: false},
  {value: '8', display: 'test 9', someOtherProp: false},
  {value: '9', display: 'test 10', someOtherProp: false},
  {value: '10', display: 'test 11', someOtherProp: false},
  {value: '11', display: 'test 12', someOtherProp: false},
  {value: '12', display: 'test 13', someOtherProp: false},
  {value: '13', display: 'test 14', someOtherProp: false},
  {value: '14', display: 'test 15', someOtherProp: false},
  {value: '15', display: 'test 16', someOtherProp: false}
]

const ExampleSelectFieldDefault = () => (
  <SelectField
    options={options}
    valueProp='value'
    displayProp='display'
    value={null }/>
)

export default ExampleSelectFieldDefault
