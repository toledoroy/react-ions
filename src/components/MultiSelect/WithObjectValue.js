import React, { Component } from 'react'
import { array, string, func } from 'prop-types'

export const WithObjectValue = WrappedComponent => class extends Component {
  constructor(props) {
    super(props)
  }

  static props = {
    changeCallback: func,
    options: array,
    value: array,
    valueProp: string
  }

  // Convert full object values to shallow array based on value prop
  flatten = (inflatedValue = []) => inflatedValue.map(v => v[this.props.valueProp])

  // Convert to shallow array to full object array by finding each valueProp in the options
  inflate = (flatValue = []) => {
    let inflatedValue = []

    for (let index = 0; index < this.props.options.length; index++) {
      const currentOption = this.props.options[index]

      // Push full option into inflatedValue if the value exists in the flatValue array
      if (flatValue.includes(currentOption[this.props.valueProp])) {
        inflatedValue.push(currentOption)
      }

      // Stop loop if all values have been found
      if (inflatedValue.length === flatValue.length) return inflatedValue
    }

    return inflatedValue
  }

  handleChanges = event => {
    if (typeof this.props.changeCallback === 'function') {
      this.props.changeCallback({
        ...event,
        value: this.inflate(event.target.value)
      })
    }
  }

  render = () => {
    const { value, changeCallback, ...otherProps } = this.props
    const flatValue = this.flatten(value)

    return <WrappedComponent value={flatValue} changeCallback={this.handleChanges} {...otherProps} />
  }
}

export default WithObjectValue
