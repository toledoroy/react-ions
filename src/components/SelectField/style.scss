@import '../../styles/common/colors';
@import '../../styles/common/screen';
@import '../../styles/common/mixins';

.selectfield-component {
  display: inline-block;
  position: relative;

  label {
    display: block;
    margin-bottom: 10px;
    font-size: 16px;
    color: $navy-text;
    font-weight: 200;
  }

  .selectfield-value {
    align-items: center;
    background-color: $white;
    border: 2px solid $neutral-3;
    border-radius: 3px;
    box-sizing: border-box;
    color: $primary-4;
    cursor: pointer;
    background: $white;
    display: flex;
    position: relative;
    padding: 10px;
    justify-content: space-between;
    user-select: none;
    transition: border-color ease-in-out .15s;

    > svg:last-of-type {
      transform: rotate(0);
    }

    > .icon {
      padding-right: 8px;
    }

    &:hover {
      border-color: rgba($neutral-4, .4);
    }

    &:focus {
      border-color: rgba($neutral-4, .6);
      outline: 0;
    }
    
    @media #{$media-extra-small} {
      width: 100% !important;
    }
  }

  .display-text {
    display: block;
    overflow: hidden;
    padding-right: 8px;
    text-overflow: ellipsis;
    white-space: nowrap;
  }

  ul {
    background-color: $white;
    border-radius: 3px;
    box-shadow: 0 1px 2px 1px rgba(0, 0, 0, 0.2);
    box-sizing: border-box;
    display: none;
    left: 0;
    margin: 3px 0 30px;
    overflow: auto;
    padding: 0;
    position: absolute;
    width: 100%;
    max-height: 152px;
    z-index: 3;

    li {
      @include vertical-align('block');
      color: $primary-4;
      cursor: pointer;
      display: flex;
      padding: 10px;
      text-decoration: none;
      min-height: 18px;

      svg {
        margin-right: 10px;
      }

      &:not(.not-clickable):hover {
        svg {
          fill: $white;
        }
        background-color: $primary-1;
        color: $white;
      }

      &.not-clickable {
        cursor: default;
      }

      &.hidden {
        display: none;
      }
    }

    @media #{$media-small-minus} {
      left: 0;
      min-width: 100%;
    }
    
    @media #{$media-extra-small} {
      width: 100% !important;
    }
  }

  &.selectfield-disabled {
    cursor: not-allowed;

    > div {
      background-color: rgba($neutral-3, .6);
      border-color: $neutral-3;
      color: rgba($primary-4, 0.5);
      cursor: not-allowed;

      > svg {
        fill: $neutral-4;
      }
    }

    > ul {
      display: none;
    }

    &:hover .selectfield-value {
      border-color: $neutral-3;
    }
  }

  &.selectfield-error {
    > div {
      border-color: $danger;
    }
  }

  &.active:not(.selectfield-disabled) {
    .selectfield-value {
      border-color: rgba($neutral-4, .6);

      > svg:last-of-type {
        transform: rotate(180deg);
      }
    }
    ul {
      display: inline-block;
    }
  }

  &.fill {
    width: 100%;
  }

  @media #{$media-small-minus} {
    min-width: 100%;
  }
}
