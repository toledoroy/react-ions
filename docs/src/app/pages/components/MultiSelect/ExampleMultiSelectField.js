import React from 'react'
import MultiSelect from 'react-ions/lib/components/MultiSelect/MultiSelect'
import SelectField from 'react-ions/lib/components/SelectField/SelectField'
import Button from 'react-ions/lib/components/Button'
import style from './style.scss'

class ExampleMultiSelectField extends React.Component {
  constructor(props) {
    super(props)
  }

  state = {
    value: ['0', '1'],
    status: 'Chosen options are: 0,1'
  }

  handleChange = event => {
    this.setState({ value: event.target.value, status: 'The callback was triggered and ' + (event.target.value.length > 0 ? 'the chosen options are: ' + event.target.value.join(',') : 'there are no options selected') })
  }

  updateSelected = index => {
    this.setState({ value: [index], status: 'Chosen options are: ' + index })
  }

  render() {
    const options = [
      {value: '0', display: 'Test 1'},
      {value: '1', display: 'Test 2'},
      {value: '2', display: 'Test 3'},
      {value: '3', display: 'Test 4'}
    ]

    return (
      <div>
        <div className={style.update}>
          <Button onClick={this.updateSelected.bind(this, '0')}>Select 1st item</Button>
        </div>
        <MultiSelect optClass={style['custom-multi']} options={options} valueProp='value' displayProp='display' value={this.state.value} changeCallback={this.handleChange} placeholder="Select one or more items">
          <SelectField />
        </MultiSelect>
        <code className={style['callback-status']}>{this.state.status}</code>
      </div>
    )
  }
}

export default ExampleMultiSelectField
