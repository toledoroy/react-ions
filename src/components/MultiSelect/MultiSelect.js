import React from 'react'
import PropTypes from 'prop-types'
import optclass from '../internal/OptClass'
import TagList from '../internal/TagList'
import style from './style.scss'

class MultiSelect extends React.Component {
  constructor(props) {
    super(props)
  }

  static defaultProps = {
    disabled: false
  }

  static propTypes = {
    /**
     * A string to display as the placeholder text.
     */
    placeholder: PropTypes.string,
    /**
     * An array of objects which will be used as the options for the MultiSelect component.
     */
    options: PropTypes.array.isRequired,
    /**
     * The values of the options to be selected.
     */
    value: PropTypes.array,
    /**
     * Which field in the option object will be used as the value of the MultiSelect component.
     */
    valueProp: PropTypes.string.isRequired,
    /**
     * Which field in the option object will be used as the display of the MultiSelect component.
     */
    displayProp: PropTypes.string.isRequired,
    /**
     * Whether the MultiSelect component is disabled.
     */
    disabled: PropTypes.bool,
    /**
     * A callback function to be called when an option is selected.
     */
    changeCallback: PropTypes.func,
    /**
     * Optional CSS class(es) to be used for local styles (string or array of strings)
     */
    optClass: PropTypes.oneOfType([
      PropTypes.array,
      PropTypes.string
    ]),
    /**
     * A customized list of selected items.
     */
    tagList: PropTypes.oneOfType([
      PropTypes.func,
      PropTypes.node
    ])
  }

  state = {
    isOpen: false,
    value: this.props.value || []
  }

  UNSAFE_componentWillMount = () => {
    // Set state
    if (this.state.value instanceof Array && this.state.value.length > 0 && this.containsValidValue(this.state.value, this.props.options)) {
      this.setState({selected: this.getSelectedOptions(this.state.value), value: this.state.value})
    }
    // No value is passed in
    else {
      this.setState({selected: [], value: []})
    }
  }

  UNSAFE_componentWillReceiveProps = nextProps => {
    if (nextProps.value !== this.state.value) {
      // Set state
      if (nextProps.value instanceof Array && (this.containsValidValue(nextProps.value, nextProps.options) || nextProps.value.length === 0)) {
        this.setState({selected: this.getSelectedOptions(nextProps.value), value: nextProps.value})
      }
      // No value is passed in
      else {
        this.setState({selected: [], value: []})
      }
    }
  }

  getIndex = value => {
    let optionIndex = -1

    this.props.options.map((option, index) => {
      if (option[this.props.valueProp] === value) {
        optionIndex = index
      }
    })

    return optionIndex
  }

  containsValidValue = (values, options) => {
    let isValid = false

    for (let i = 0; i < values.length; i++) {
      if (this.getIndex(values[i], options) > -1) {
        isValid = true
      }
    }

    return isValid
  }

  getSelectedOptions = values => {
    let selectedOptions = []

    values.map((value, index) => {
      this.props.options.map((option, index) => {
        if (option[this.props.valueProp] === value) {
          selectedOptions.push(option)
        }
      })
    })

    return selectedOptions
  }

  handleChange = event => {
    // when value & option are empty it means that reset button was clicked
    if (event.target.value !== '' && event.target.option !== '') {
      let values = this.state.value

      values.push(event.target.value)

      this.setState({selected: this.getSelectedOptions(values), value: values}, () => {
        if (this.props.changeCallback) {
          this.props.changeCallback({
            target: {
              name: this.props.name,
              value: this.state.value,
              options: this.state.selected
            }
          })
        }
      })
    }
  }

  filterOptions = option => {
    const optionValue = option[this.props.valueProp]

    return (this.state.value.indexOf(optionValue) === -1)
  }

  getElements = children => {
    let {options, value, ...props} = this.props

    props.options = this.props.options.filter(this.filterOptions)
    props.changeCallback = this.handleChange

    if (Array.isArray(children)) {
      children = children[0]
    }

    if (['WrappedTypeahead', 'Wrappedt'].includes(children.type.displayName)) {
      props.resetAfterSelection = true
      props.optionsFilterPredicate = this.filterOptions
    }

    return React.Children.map(children, child => {
      let {options, value, ...childProps} = child.props

      return React.cloneElement(child, Object.assign(childProps, props))
    })
  }

  onRemove = index => {
    const values = this.state.value.slice()

    values.splice(index, 1)

    this.setState({selected: this.getSelectedOptions(values), value: values}, () => {
      if (this.props.changeCallback) {
        this.props.changeCallback({
          target: {
            name: this.props.name,
            value: this.state.value,
            options: this.state.selected
          }
        })
      }
    })
  }

  render() {
    const multiSelectClasses = optclass(style, 'multi-select', this.props.optClass)
    const elements = this.getElements(this.props.children)
    const TagListComponent = this.props.tagList || TagList

    return (
      <div className={multiSelectClasses}>
        {elements}
        <TagListComponent tags={this.state.selected} displayProp={this.props.displayProp} onRemove={this.onRemove} />
      </div>
    )
  }
}

export default MultiSelect
