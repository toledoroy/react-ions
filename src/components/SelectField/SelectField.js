import React from 'react'
import PropTypes from 'prop-types'
import enhanceWithClickOutside from 'react-click-outside'
import classNames from 'classnames/bind'
import style from './style.scss'
import Icon from '../Icon'

export class SelectField extends React.Component {
  constructor(props) {
    super(props)
  }

  static defaultProps = {
    disabled: false,
    options: [],
    valueProp: '',
    displayProp: ''
  }

  static propTypes = {
    /**
     * A string to display as the placeholder text.
     */
    placeholder: PropTypes.string,
    /**
     * An array of objects which will be used as the options for the select field.
     */
    options: PropTypes.array.isRequired,
    /**
     * The value of the option to be selected.
     */
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    /**
     * Which field in the option object will be used as the value of the select field.
     */
    valueProp: PropTypes.string.isRequired,
    /**
     * Which field in the option object will be used as the display of the select field.
     */
    displayProp: PropTypes.string.isRequired,
    /**
     * Which field in the option object will be used to determine whether the option should be hidden.
     */
    hideProp: PropTypes.string,
    /**
     * Whether the select field is disabled.
     */
    disabled: PropTypes.bool,
    /**
     * Optional styles to add to the select field.
     */
    optClass: PropTypes.string,
    /**
     * A callback function to be called when an option is selected.
     */
    changeCallback: PropTypes.func,
    /**
     * Icon to be displayed on the left
     */
    icon: PropTypes.string,
    /**
     * Text shown above the select field.
     */
    label: PropTypes.string,
    /**
     * An optional style object to render inline CSS.
     */
    style: PropTypes.object,
    /**
     * A helper will render inline style='width: <value>'.
     */
    width: PropTypes.string
  }

  state = {
    isOpen: false,
    value: this.props.value
  }

  componentWillMount = () => {
    // Select item
    if (this.state.value !== '' && this.getIndex(this.state.value, this.props.options) > -1) {
      this.selectItem(this.state.value, this.props.options)
    }
    // No value is passed in
    else {
      this.setState({selected: '', value: ''})
    }
  }

  componentWillReceiveProps = nextProps => {
    if (nextProps.value === this.state.value) {
      return
    }
    let validIndex = this.getIndex(nextProps.value, nextProps.options) > -1

    if (nextProps.value === undefined || nextProps.value === null || !validIndex) {
      this.setState({selected: '', value: ''})
    } else {
      this.selectItem(nextProps.value, nextProps.options)
    }
  }

  handleClickOutside = () => {
    if (this.state.isOpen) this.toggleSelectField()
  }

  toggleSelectField = () => {
    if (this.props.disabled) return

    this.setState({isOpen: !this.state.isOpen})
  }

  selectOption = (option, triggerCallback) => {
    this.setState({ selected: option, value: option[this.props.valueProp], isOpen: false }, () => {
      if (triggerCallback && typeof this.props.changeCallback === 'function') {
        this.props.changeCallback({
          target: {
            name: this.props.name,
            value: option[this.props.valueProp],
            option: option
          }
        })
      }
    })
  }

  selectItem = (value, options) => {
    let index = this.getIndex(value, options)

    if (index >= 0) {
      this.selectOption(options[index], false)
    }
  }

  getIndex = (value, options) => {
    let optionIndex = -1

    options.map((option, index) => {
      if (option[this.props.valueProp] === value) {
        optionIndex = index
      }
    })

    return optionIndex
  }

  getDisplayText = () => {
    if (this.state.selected !== '') {
      return this.state.selected[this.props.displayProp]
    } else if (typeof this.props.placeholder !== 'undefined') {
      return this.props.placeholder
    }
    return 'Please select an option'

  }

  getDisplayIcon = () => {
    if (this.state.selected && this.state.selected.icon) {
      return <Icon name={this.state.selected.icon} fill={this.state.selected.iconColor || null} className={style.icon} height='16' width='16' />
    } else if (this.props.icon) {
      return <Icon name={this.props.icon} className={style.icon} height='16' width='16' />
    }
    return null

  }

  render() {
    const cx = classNames.bind(style)
    const disabledClass = this.props.disabled ? style['selectfield-disabled'] : ''
    const activeClass = this.state.isOpen ? style['active'] : ''
    const selectFieldClass = cx(style['selectfield-component'], activeClass, disabledClass, this.props.optClass)
    const { valueProp, hideProp, label, width } = this.props
    
    const widthStyle = width ? { width: width } : null
    const listWidthStyle =  width ? { width: width, minWidth: width } : null

    let options = this.props.options.map((option, index) =>
      <li
        key={index}
        onClick={this.selectOption.bind(null, option, true)}
        className={(hideProp && option[hideProp]) ? style['hidden'] : undefined}>
          {option.icon
            ? <Icon
              name={option.icon}
              fill={option.iconColor || null}
              className={style.icon}
              height='16'
              width='16' />
            : null
          }
          {option[this.props.displayProp]}
        </li>
    )

    if (options.length === 0) {
      options.push(<li key={0} className={style['not-clickable']}>Nothing to select</li>)
    }

    let value = ''

    if (this.state.selected) {
      value = this.state.selected[valueProp]
    }

    return (
      <div className={selectFieldClass} style={this.props.style}>
        <input type='hidden' name='selectfield-value' value={value} />
        { label && <label>{label}</label> }
        <div className={style['selectfield-value']} onClick={this.toggleSelectField} style={widthStyle}>
          {this.getDisplayIcon()}
          <span className={style['display-text']}>{this.getDisplayText()}</span>
          <Icon name='md-keyboard-down' width='16' height='16' />
        </div>
        <ul style={listWidthStyle}>
          {options}
        </ul>
      </div>
    )
  }
}

export default enhanceWithClickOutside(SelectField)
