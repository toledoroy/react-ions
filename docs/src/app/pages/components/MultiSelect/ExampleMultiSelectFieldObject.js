import React from 'react'
import WithObjectValue from 'react-ions/lib/components/MultiSelect/WithObjectValue'
import MultiSelect from 'react-ions/lib/components/MultiSelect/MultiSelect'
import SelectField from 'react-ions/lib/components/SelectField/SelectField'
import style from './style.scss'

const value = [
  {value: '2', display: 'Test 2'},
  {value: '3', display: 'Test 3'}
]

const WithObjectValueDemo = WithObjectValue(MultiSelect)

class ExampleMultiSelectFieldObject extends React.Component {
  constructor(props) {
    super(props)
  }

  handleChange = value => {
    console.log(value.target.options)
  }

  render() {
    const options = [
      {value: '1', display: 'Test 1'},
      {value: '2', display: 'Test 2'},
      {value: '3', display: 'Test 3'},
      {value: '4', display: 'Test 4'},
      {value: '5', display: 'Test 5'},
      {value: '6', display: 'Test 6'},
      {value: '7', display: 'Test 7'},
      {value: '8', display: 'Test 8'},
      {value: '9', display: 'Test 9'},
      {value: '10', display: 'Test 10'}
    ]

    return (
      <div>
        <WithObjectValueDemo
          optClass={style['custom-multi']}
          options={options}
          valueProp='value'
          name='example'
          displayProp='display'
          value={value}
          changeCallback={this.handleChange}
          placeholder="Select one or more items">
          <SelectField />
        </WithObjectValueDemo>
      </div>
    )
  }
}

export default ExampleMultiSelectFieldObject
