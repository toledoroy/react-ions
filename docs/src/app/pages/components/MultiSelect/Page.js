import React from 'react'
import PropsList from 'private/modules/PropsList'
import docs from '!!docgen!react-ions/lib/components/MultiSelect/MultiSelect'
import CodeExample from 'private/modules/CodeExample'
import ExampleMultiSelectField from './ExampleMultiSelectField'
import exampleMultiSelectFieldCode from '!raw!./ExampleMultiSelectField'
import ExampleMultiTypeahead from './ExampleMultiTypeahead'
import exampleMultiTypeaheadCode from '!raw!./ExampleMultiTypeahead'
import ExampleMultiTypeaheadSearch from './ExampleMultiTypeaheadSearch'
import exampleMultiTypeaheadSearchCode from '!raw!./ExampleMultiTypeaheadSearch'
import ExampleMultiSelectFieldObject from './ExampleMultiSelectFieldObject'
import exampleMultiSelectFieldObjectCode from '!raw!./ExampleMultiSelectFieldObject'

import styles from 'private/css/content'

const description = {
  multiSelectField: 'This is the `multi select field component`.',
  multiTypeahead: 'This is the `multi typeahead component`.',
  multiTypeaheadSearch: 'This is the `multi typeahead component` with `searchCallback`.',
  multiSelectFieldObject: 'This is the `multi select field component` with an HOC that returns the original selected object, instead of the value.',
}

const SelectFieldPage = () => (
  <div>
    <div className={styles.content}>
      <div className={styles.block}>
        <CodeExample
          title='Multi Select Field'
          description={description.multiSelectField}
          markup={exampleMultiSelectFieldCode}>
          <ExampleMultiSelectField />
        </CodeExample>
        <CodeExample
          title='Multi Typeahead'
          description={description.multiTypeahead}
          markup={exampleMultiTypeaheadCode}>
          <ExampleMultiTypeahead />
        </CodeExample>
        <CodeExample
          title='Multi Typeahead with searchCallback'
          description={description.multiTypeaheadSearch}
          markup={exampleMultiTypeaheadSearchCode}>
          <ExampleMultiTypeaheadSearch />
        </CodeExample>
        <CodeExample
          title='Multi Typeahead with return object value'
          description={description.multiSelectFieldObject}
          markup={exampleMultiSelectFieldObjectCode}>
          <ExampleMultiSelectFieldObject />
        </CodeExample>
        <div className={styles.block}>
          <h3>Props</h3>
          <PropsList list={docs[0].props} />
        </div>
      </div>
    </div>
  </div>
)

export default SelectFieldPage
