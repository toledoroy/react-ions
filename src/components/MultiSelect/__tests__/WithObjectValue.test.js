import React from 'react'
import MultiSelect from '../MultiSelect'
import SelectField from '../../SelectField/SelectField'
import WithObjectValue from '../WithObjectValue'

const WithObjectValueDemo = WithObjectValue(MultiSelect)

const value = [
  {value: '2', display: 'test 2'},
  {value: '3', display: 'test 3'}
]

const options = [
  {value: '0', display: 'test 1'},
  {value: '1', display: 'test 2'},
  {value: '2', display: 'test 3'},
  {value: '3', display: 'test 4'}
]

describe('WithObjectValue', () => {
  let multiSelect

  beforeEach(() => {
    multiSelect = shallow(
      <WithObjectValueDemo
        options={options}
        valueProp='value'
        name='example'
        displayProp='display'
        value={value}
        placeholder="Select one or more items">
        <SelectField />
      </WithObjectValueDemo>
    )
  })

  it('should render', () => {
    expect(multiSelect).toMatchSnapshot()
  })

  it('should return an empty list', () => {
    expect(multiSelect.instance().flatten()).toMatchSnapshot()
  })

  it('should return a flattened list when passed a list of objects', () => {
    expect(multiSelect.instance().flatten(options)).toMatchSnapshot()
  })

  it('should return an inflated list when passed a flatted list', () => {
    expect(multiSelect.instance().inflate(['0', '1', '2', '3'])).toMatchSnapshot()
  })
})
