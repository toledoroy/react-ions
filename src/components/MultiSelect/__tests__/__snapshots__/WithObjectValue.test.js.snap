// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithObjectValue should render 1`] = `
<MultiSelect
  changeCallback={[Function]}
  disabled={false}
  displayProp="display"
  name="example"
  options={
    Array [
      Object {
        "display": "test 1",
        "value": "0",
      },
      Object {
        "display": "test 2",
        "value": "1",
      },
      Object {
        "display": "test 3",
        "value": "2",
      },
      Object {
        "display": "test 4",
        "value": "3",
      },
    ]
  }
  placeholder="Select one or more items"
  value={
    Array [
      "2",
      "3",
    ]
  }
  valueProp="value"
>
  <WrappedSelectField />
</MultiSelect>
`;

exports[`WithObjectValue should return a flattened list when passed a list of objects 1`] = `
Array [
  "0",
  "1",
  "2",
  "3",
]
`;

exports[`WithObjectValue should return an empty list 1`] = `Array []`;

exports[`WithObjectValue should return an inflated list when passed a flatted list 1`] = `
Array [
  Object {
    "display": "test 1",
    "value": "0",
  },
  Object {
    "display": "test 2",
    "value": "1",
  },
  Object {
    "display": "test 3",
    "value": "2",
  },
  Object {
    "display": "test 4",
    "value": "3",
  },
]
`;
